processor/processor.go

```diff
@@ -2529,6 +2529,34 @@ func (proc *Handle) sendQueryRetryStats(attempt int) {
 }
 
 func (proc *Handle) storeStage(partition string, in *storeMessage) {
+	lockRouterDestIDs := func() func() {
+		var deferred []func()
+		if len(in.destJobs) > 0 {
```

Comment:
[optional] We can avoid a bit of nesting since `deferred` would be empty with no jobs.

```go
if len(in.destJobs) == 0 {
    return func(){}
}
```